server/conn.go

```diff
@@ -1441,18 +1441,26 @@ func (cc *clientConn) flush(ctx context.Context) error {
 }
 
 func (cc *clientConn) writeOK(ctx context.Context) error {
-	msg := cc.ctx.LastMessage()
-	return cc.writeOkWith(ctx, msg, cc.ctx.AffectedRows(), cc.ctx.LastInsertID(), cc.ctx.Status(), cc.ctx.WarningCount(), mysql.OKHeader)
+	return cc.writeOkWith(ctx, true, true, cc.ctx.Status())
 }
 
-func (cc *clientConn) writeOkWith(ctx context.Context, msg string, affectedRows, lastInsertID uint64, status, warnCnt uint16, header byte) error {
+func (cc *clientConn) writeOkWith(ctx context.Context, okOReof bool, flush bool, status uint16) error {
```

Comment:
In my [PR](https://github.com/pingcap/tidb/pull/34753) I changed the signature to have the `header byte` instead of a bool as suggested by @bb7133. This is also what https://github.com/pingcap/tidb/pull/32119 did.

Both are fine with me. However I would suggest to name this `eof bool` to make it clear that `true`  == EOF.

Reply:
Sound good to me.

Reply:
I may have a try on `header byte` after CI passed.

Reply:
Done